File: 백준/Silver/1182. 부분수열의 합/부분수열의 합.java
import java.util.*;
import java.io.*;

public class Main {
	static int count = 0;
	static int[] arr;
	static int n, s;

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		n = Integer.parseInt(st.nextToken());
		s = Integer.parseInt(st.nextToken());
		st = new StringTokenizer(br.readLine());
		arr = new int[n];
		for (int i = 0; i < n; i++) {
			arr[i] = Integer.parseInt(st.nextToken());
		}
		dfs(0,0);
		System.out.println(count);
	}

	public static void dfs(int idx, int sum) {
		if (idx == n)
			return;
		if( sum+ arr[idx] == s)
			count++;
		dfs(idx+1,sum);
		dfs(idx+1, sum+arr[idx]);

	}

}

File: 백준/Silver/1182. 부분수열의 합/부분수열의 합.py

# N 정수의 개수   1<=n <=20
# 목표 합 S 
def dfs(idx, sum) :
	global count
	if idx == n:
		return
	if sum+ num_list[idx] == s:
		count+=1
	
    #들어온 index 포함x
	dfs(idx+1,sum)
	
    #들어온 index 포함
	dfs(idx+1, sum+num_list[idx])

n,s = map(int,input().split(" "))
num_list = list(map(int,input().split()))
count = 0
dfs(0,0)

print(count)